README.md

```diff
@@ -59,6 +59,36 @@ See the [git documentation](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-H
 
 Most code may be tested with vanilla Go unit tests.
 
+## shell scripts unit tests
+
+Please visit the official [GitHub link](https://github.com/shellspec/shellspec) for more details. Below is a brief use case.
+
+### Installation 
+
+`Shellspec` is used as a framework for unit test.
+
+To run it, please install it by using `curl -fsSL https://git.io/shellspec | sh`.
+
+By default, it should be installed in `~/.local/lib/shellspec`. Please append it to the $PATH for your convenience. Example command `export PATH=$PATH:~/.local/lib/shellspec`.
+
+### Authoring tests
+
+You will need to write `xxx_spec.sh` file for the test.
+
+For example, `AgentBaker/spec/parts/linux/cloud-init/artifacts/cse_install_spec.sh` is a test file for `AgentBaker/parts/linux/cloud-init/artifacts/cse_install.sh`
+
+### Running tests locally
+
+To run all tests, in AgentBaker folder, simply run `bash shellspec`. (Assuming you set the $PATH correctly, otherwise you will need to provide the full path for shellspec binary)
+
+#### Useful commands for debugging

```

Comment:
could we incorporate installation/usage into the Makefile we already have to keep things simple?

we'd probably want to incorporate this into the `generate` target anyways

Reply:
Will do! If time does not allow, I will do it in the next PR. (Just trying to unblock other folks with this PR)

Reply:
done